File: src/main/java/com/jackmeng/stl/stl_Bezier.java
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package com.jackmeng.stl;

import java.util.List;

public class stl_Bezier
{
    private final List< Double > controlPoints;
    private final int degree;

    public stl_Bezier(List< Double > controlPoints, int degree)
    {
        this.controlPoints = controlPoints;
        this.degree = degree;
    }

    public double interpolate(double t)
    {
        double result = 0;
        for (int i = 0; i <= degree; i++)
        {
            double controlPoint = controlPoints.get(i);
            double coefficient = stl_Maths.binomial_coefficient(degree, i) * Math.pow(1 - t, (double) degree - i)
                    * Math.pow(t, i);
            result += controlPoint * coefficient;
        }
        return result;
    }

}

File: src/main/java/com/jackmeng/stl/stl_Bezier.kt
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
package com.jackmeng.stl

import com.jackmeng.stl.stl_Maths

class stl_Bezier(private val controlPoints:List<Double> , private val degree:Int)
{
	fun interpolate(t:Double):Double
	{
		var result=0.0
		for (i in 0..degree)
		{
			val controlPoint=controlPoints[i]
			val coefficient=(stl_Maths.binomial_coefficient(degree , i)*Math.pow(1-t , degree.toDouble()-i)*Math.pow(
				t ,
				i.toDouble()
			))
			result+=controlPoint*coefficient
		}
		return result
	}
}